src/test/java/io/temporal/workflow/MetricsTest.java

```diff
@@ -335,4 +354,63 @@ public WorkflowInvoker interceptExecuteWorkflow(
       return new BaseWorkflowInvoker(i, next);
     }
   }
+
+  @Test
+  public void testTemporalFailureMetric() throws InterruptedException {
+    setUp(
+        com.uber.m3.util.Duration.ofMillis(300),
+        WorkerFactoryOptions.newBuilder()
+            .setWorkflowInterceptor(new CorruptedSignalWorkflowInterceptor())
+            .build());
+
+    try {
+      WorkflowServiceStubs serviceStubs =
+          testEnvironment.getWorkflowClient().getWorkflowServiceStubs();
+
+      serviceStubs.blockingStub().describeNamespace(DescribeNamespaceRequest.newBuilder().build());
+      fail("failure expected");
+    } catch (StatusRuntimeException e) {
+      assertEquals(Status.Code.UNIMPLEMENTED, e.getStatus().getCode());
+    }
+
+    // Wait for reporter
+    Thread.sleep(600);
+
+    verify(reporter, times(1))
+        .reportCounter("temporal-DescribeNamespace.temporal-request", new HashMap<>(), 1);
+    verify(reporter, times(1))
+        .reportCounter("temporal-DescribeNamespace.temporal-error", new HashMap<>(), 1);
+    testEnvironment.close();
+  }
+
+  @Test
+  public void testTemporalInvalidRequestMetric() throws InterruptedException {
+    setUp(
+        com.uber.m3.util.Duration.ofMillis(300),
+        WorkerFactoryOptions.newBuilder()
+            .setWorkflowInterceptor(new CorruptedSignalWorkflowInterceptor())
+            .build());
+
+    try {
+      WorkflowServiceStubs serviceStubs =
+          testEnvironment.getWorkflowClient().getWorkflowServiceStubs();
+
+      serviceStubs
+          .blockingStub()
+          .startWorkflowExecution(StartWorkflowExecutionRequest.newBuilder().build());
+      fail("failure expected");
+    } catch (StatusRuntimeException e) {
+      assertEquals(Status.Code.INVALID_ARGUMENT, e.getStatus().getCode());
+    }
+
+    // Wait for reporter
+    Thread.sleep(600);
+
+    verify(reporter, times(1))
+        .reportCounter("temporal-StartWorkflowExecution.temporal-request", new HashMap<>(), 1);
+    verify(reporter, times(1))
+        .reportCounter(
+            "temporal-StartWorkflowExecution.temporal-invalid-request", new HashMap<>(), 1);
```

Comment:
Are these defined somewhere as consts or is there a common way to generate them?

Reply:
Replaced with constants